docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/pattern/helper/PropertyValueHelper.java

```diff
@@ -153,35 +172,77 @@ public static String getPVGTable (CapellaElement element, int level, String proj
 			} else {
 				grelation += "Applied";
 			}
-			
-			result.append("<tr>");
-				result.append("<td><i>");
-					result.append(grelation);
-				result.append("</i></td>");
-				result.append("<td>");
-					result.append(gname);
-				result.append("</td>");
-				result.append("<td>");
-					result.append(gvalue);
-				result.append("</td>");
-				result.append("<td>");
-					result.append(gdescription);
-				result.append("</td>");
-			result.append("</tr>");
-			
+
+			result.append(getPVRow(grelation, gname, gvalue, gdescription));
+
 			alreadyGeneratedElements.add(element);
 
 			CapellaElement subelement = (CapellaElement) group;
 			if (!alreadyGeneratedElements.contains(subelement)) {
-				//We deal with its owned / applied Property Values
+				// We deal with its owned / applied Property Values
 				String tablePV = PropertyValueHelper.getPVTable(subelement, level + 1, projectName, outputFolder);
 				result.append(tablePV);
-				
-				//We deal with its owned / applied Property Value Groups
-				String tablePVG = PropertyValueHelper.getPVGTable(subelement, level + 1, projectName, outputFolder, alreadyGeneratedElements);
+
+				// We deal with its owned / applied Property Value Groups
+				String tablePVG = PropertyValueHelper.getPVGTable(subelement, level + 1, projectName, outputFolder,
+						alreadyGeneratedElements);
 				result.append(tablePVG);
 			}
 		}
 		return result.toString();
 	}
+
+	private static String getPVRow(String relation, StringBuffer name, String value, String description) {
+		StringBuffer result = new StringBuffer();
+		result.append("<tr>");
+		result.append("<td><i>");
+		result.append(relation);
+		result.append("</i></td>");
+		result.append("<td>");
+		result.append(name);
+		result.append("</td>");
+		result.append("<td>");
+		result.append(value);
+		result.append("</td>");
+		result.append("<td>");
+		result.append(description);
+		result.append("</td>");
+		result.append("</tr>");
+		return result.toString();
+	}
+
+	private static boolean isInSystemEngineering(EObject elem) {
```

Comment:
What do you think about this implementation:

	private static boolean isInSystemEngineering(EObject elem) {
		EObject container = elem.eContainer();
		if (container != null) {
			return container instanceof SystemEngineering ? true : isInSystemEngineering(container);
		}
		return false;
	}

Reply:
Yes !